packages/@lwc/synthetic-shadow/src/shared/node-collection-util.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { isUndefined, isTrue, ArrayPush } from './language';
+/**
+ * Writing our own utils to handle NodeList and HTMLCollection. This is to not conflict with
+ * some legacy third party libraries like prototype.js that patch Array.prototype.
+ */
+
+/**
+ * Custom implementation of filter since using Array.prototype.filter conflicts with other
+ * legacy libraries like prototype.js
+ * Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter#Polyfill
+ */
+export function collectionFilter<T extends Node, K extends Element>(
+    collection: NodeListOf<T> | HTMLCollectionOf<K>,
+    fn: (value: T | K, index?: number, collection?: NodeListOf<T> | HTMLCollectionOf<K>) => boolean
+): Array<T | K> {
+    const res: Array<T | K> = [];
+    const length = collection.length;
+    for (let i = 0; i < length; i++) {
+        const curr = collection[i];
+        if (isTrue(fn(curr, i, collection))) {
+            ArrayPush.call(res, curr);
+        }
+    }
+    return res;
+}
+
+/**
+ * Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find#Polyfill
+ */
+export function collectionFind<T extends Node>(
+    collection: NodeListOf<T>,
+    fn: (value: T, index?: number, nodelist?: NodeListOf<T>) => boolean
+): T | undefined {
+    const length = collection.length;
+    for (let i = 0; i < length; i++) {
+        const curr = collection[i];
+        if (isTrue(fn(curr, i, collection))) {
```

Comment:
This check seems too strict since the spec only requires the callback return value to be coercible to a boolean:
https://tc39.es/ecma262/#sec-array.prototype.find